src/details/ArborX_DetailsDistributedTreeImpl.hpp

```diff
@@ -62,21 +62,21 @@ struct DistributedTreeImpl
                 IndicesAndRanks &values, Offset &offset);
 
   // nearest neighbors helpers
-  template <typename DistributedTree, typename ExecutionSpace,
-            typename Predicates, typename Indices, typename Offset,
-            typename Distances>
+  template <typename ExecutionSpace, typename DistributedTree,
+            typename Predicates, typename Distances, typename Indices,
+            typename Offset>
   static void deviseStrategy(ExecutionSpace const &space,
-                             Predicates const &queries,
-                             DistributedTree const &tree, Indices &indices,
-                             Offset &offset, Distances &);
+                             DistributedTree const &tree,
+                             Predicates const &queries, Distances const &,
```

Comment:
Why did you do change
(indices, offsets, distances) -> (distances, indices, offsets)
and why did you const the distances?

Reply:
We only read the distances, and don't modify them. So `const` makes it clearer.
I changed the order to follow the same principle: input arguments followed by output arguments.